riscv-page-tables/src/page_table.rs

```diff
@@ -624,8 +632,7 @@ impl<T: PagingMode> PageTableInner<T> {
 
 /// A paging hierarchy for a given addressing type.
 /// This is used for HS and HU paging modes such as Sv48.
-///
-/// TODO: Support non-4k page sizes.
```

Comment:
Not sure this TODO is ready to come out yet -- this patch adds support for _mapping_ huge-pages, but we still need to support unmap and invalidation.